driver-sync/src/test/functional/com/mongodb/client/ChangeStreamProseTest.java

```diff
@@ -401,6 +405,40 @@ public void testGetResumeTokenReturnsIdOfPreviousDocument() {
         }
     }
 
+    @Test
+    public void test19SplitChangeStreamEvents() {
+        assumeTrue(serverVersionAtLeast(7, 0));
+        collection.drop();
+        database.createCollection(
+                getClass().getName(),
+                new CreateCollectionOptions().changeStreamPreAndPostImagesOptions(
+                        new ChangeStreamPreAndPostImagesOptions(true)));
+
+        // #. Insert into _C_ a document at least 10mb in size ...
+        int mb10 = 10 * 1024 * 1024;
+        collection.insertOne(new Document("value", new BsonString(repeat(mb10, "q"))));
+
+        // #. Create a change stream _S_ by calling ``watch`` on _C_ with
+        //    pipeline ``[{ "$changeStreamSplitLargeEvent": {} }]`` and ``fullDocumentBeforeChange=required``.
+        ChangeStreamIterable<Document> changeStream = collection
+                .watch(asList(Document.parse("{ $changeStreamSplitLargeEvent: {} }")))
```

Comment:
I did not create a stage builder, since we do not seem to offer stages for other stages like this (for example, $currentOp).